test/suites/CurlClientTest.php

```diff
@@ -25,29 +25,69 @@
  */
 class CurlClientTest extends HttpClientTestsCase
 {
+    const FAKE_PROXY_HOST = '1111.0.0.0';
+    const FAKE_PROXY_PORT = 50;
+    const FAKE_PROXY_USER = 'user';
+    const FAKE_PROXY_PASSWORD = 'password';
+    const FAKE_PROXY_TYPE = 'HTTP';
+    const FAKE_PROXY_TYPE_CURL = CURLPROXY_HTTP;
+
     public function getHttpClient()
     {
         return new CurlClient();
     }
 
-    public function testCurlException()
+    public function getRequest()
+    {
+        return new Request('http://httpbin.org/get', 'GET');
+    }
+
+    public function getFakeProxy()
     {
+        return new Proxy(self::FAKE_PROXY_HOST, self::FAKE_PROXY_PORT, self::FAKE_PROXY_USER, self::FAKE_PROXY_PASSWORD, self::FAKE_PROXY_TYPE);
+    }
 
+    public function testCurlException()
+    {
         $client = $this->getHttpClient();
-        $request = new Request('http://httpbin.org/get', 'GET');
 
         try {
-            $client->sendRequest($request, Proxy::createFromString('1111.0.0.0:50'));
+            $client->sendRequest($this->getRequest(), $this->getFakeProxy());
             $this->fail('no exception thrown');
         } catch (CurlClient\CurlException $e) {
             $this->assertEquals(5, $e->getCurlErrCode());
             $this->assertNotEmpty($e->getCurlErrMessage());
         }
-
     }
 
     public function testGetCurl()
     {
         $this->assertInstanceOf(CurlClient\Curl::class, $this->getHttpClient()->getCurl());
     }
+
+    public function testProxyConfiguration()
+    {
+        $client = $this->getHttpClient();
+
+        try {
+            $client->sendRequest($this->getRequest(), $this->getFakeProxy());
+            $this->fail('no exception thrown');
+        } catch (CurlClient\CurlException $e) {
+            $this->assertEquals(self::FAKE_PROXY_HOST, $client->getCurl()->getOption(CURLOPT_PROXY));
+            $this->assertEquals(self::FAKE_PROXY_PORT, $client->getCurl()->getOption(CURLOPT_PROXYPORT));
+            $this->assertEquals(self::FAKE_PROXY_USER . ':' . self::FAKE_PROXY_PASSWORD, $client->getCurl()->getOption(CURLOPT_PROXYUSERPWD));
+            $this->assertEquals(self::FAKE_PROXY_TYPE_CURL, $client->getCurl()->getOption(CURLOPT_PROXYTYPE));
+        }
+    }
+
+    public function testProxyCreateFromStr()
```

Comment:
Proxies are defined in the core repository of serps. Proxy tests also live here. 

Check: https://github.com/serp-spider/core/blob/95ebbb52f0f2becf8e220d4866d9a128ea700536/test/suites/TDD/Core/http/ProxyTest.php#L39